lib/vfscore: Fix goto label issues in preadv() and pwritev()

razvand opened the pull request:
There are two issues fixed by the two commits:
* The `out_error` label used in `preadv()` in `lib/vfscore/main.c` is not properly named. It should be `out_errno`.
* The `out_errno` label in `pwrite()` is unused.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

razvand committed:
lib/vfscore: Fix label issue (out_error -> out_errno)

The out_error jump label used by goto in lib/vfscore/main.c is not
properly named. It results in an error when using vfscore. This fix
renames it to out_errno.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

nderjung commented:
Looks good to me.  Though, maybe this PR should _also_ include a fix for:

```
/tmp/build/80754af9/built/unikraft/lib/vfscore/main.c: In function 'pwrite':
/tmp/build/80754af9/built/unikraft/lib/vfscore/main.c:397:1: warning: label 'out_errno' defined but not used [-Wunused-label]
  397 | out_errno:
      | ^~~~~~~~~
```

razvand commented:
@nderjung , OK, will do.

razvand committed:
lib/vfscore: Add missing goto statement in pwritev

The label out_errno in pwritev() is not used, though it should be, as it
is the case with preadv(). This updates pwritev() and add a missing goto
statement to the out_errno label. It makes the implementation of
pwritev() consistent with preadv().

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

razvand commented:
@nderjung , I updated the PR with the fix of the warning. I also updated (forced push) the first commit, since the label name should be `out_errno`, for consistency with the label names in other functions in `lib/vfscore/main.c`.

nderjung commented:
Thanks Razvan!

Reviewed-by: Alexander Jung <a.jung@lancs.ac.uk>

nderjung reviewed (approved):
Thanks Razvan!

Reviewed-by: Alexander Jung <a.jung@lancs.ac.uk>
